### ISSUE TITLE
ON should use regexp instead of lookup table for species percent

### ISSUE THREAD

**Issue by pedrogit** [edwardsmarc/CASFRI#678]
This duplicates some ON01 rows when LEFT JOINing them with that table.

The duplicate values is "xx0".

BTW this table should be replaced with a function using regexp functions and operators to count the number of adjacent letters and the number of adjacent numbers.

SELECT cardinality(array_agg(matches))
FROM (SELECT regexp_matches('a0cb0de0fg0hi0jk0lm0', '[a-zA-Z]+', 'g') matches) foo
INTO nb_species;

percent_multiplier = CASE WHEN ('a0cb0de0fg0hi0jk0lm0' ~ '00') THEN 1 ELSE 10 END;
percent_multiplier = CASE WHEN ('x076xx000' ~ '00') THEN 1 ELSE 10 END;

**Comment by edwardsmarc** [comment]
Agreed this could be simplified using regexp and functions. One reason I used the lookup table was to validate the codes. The species codes in ON01 are a mess, there are many different formats that appear to have been combined from various sources without any standardization.

For example most codes with a single integer should be multiplied by 10 as you suggest (e.g. 'AB 4AW 3MR 3'), but then there are codes like 'BW  95PO   5' which should not. Generally any code with 2-digit values followed by 3-digit values should just be copied. These cases are included in my lookup table. But then you get some cases where the 2-digit codes come later: 'BW 5B 3W 10' (not clear what to do here. Maybe multiply the 5 and the 3 by 10. But they don't sum to 100).

I chose to use the lookup table to select the main formats that obviously didn't have any errors. Anything that doesn't match one of these formats fails validation. We could go through these codes systematically and try and decipher how to translate each one. It would be a large task.

Here are some examples of codes that get dropped currently:

'L  7SB 3   0' - many codes have this trailing zero. It is probably an error, so in these cases the trailing zero should be removed and the remaining code processed as normal.

'PO  70BF' - an error where the second percent value is missing. In these cases we should probably return UNKNOWN_VALUE for the second percent.

'BW 5B 3W 10' - mentioned above


Currently I estimate there are only 141 invalid codes using my lookup table, and 2,604,166 valid codes in ON01:
```
SELECT tt_alphaNumericMatchTable(spcomp, 'translation', 'on_species_valid_alpha_numeric_codes') codes, count(*) 
FROM rawfri.on01 
WHERE tt_notEmpty(spcomp) 
GROUP BY codes
```
